File: python/03_repetitions.py
dna = input().strip()

longest = 1
t = 1
for i in range(1, len(dna)):
    if dna[i] == dna[i - 1]:
        t += 1
    else:
        longest = max(longest, t)
        t = 1

longest = max(longest, t)
print(longest)

File: rust/src/bin/03_repetitions.rs
use std::io;

fn main() {
    let mut dna = String::new();
    let _ = io::stdin().read_line(&mut dna);

    let dna: Vec<char> = dna.chars().collect();

    let mut longest = 1;
    let mut t = 1;
    for i in 1..dna.len() {
        if dna[i] == dna[i - 1] {
            t += 1;
        } else {
            longest = longest.max(t);
            t = 1;
        }
    }

    longest = longest.max(t);
    println!("{}", longest);
}
